crates/relayer/src/config.rs

```diff
@@ -556,14 +558,37 @@ pub enum EventSourceMode {
 
 #[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
 #[serde(deny_unknown_fields)]
-pub struct ChainConfig {
+#[serde(tag = "type")]
+pub enum ChainConfig {
+    CosmosSdk(CosmosSdkConfig),
+}
```

Comment:
As mentioned in the issue description for #3636, this will produce TOML like
```toml
[[chains]]
type = "CosmosSdk"
id = "chain_id"
rpc_addr = "http://127.0.0.1:26657/"
grpc_addr = "http://127.0.0.1:8080/"
```
The only difference from the current format is that the `type` field is now mandatory, and it must be `CosmosSdk`, not `cosmos-sdk` or `cOsMo-sSdk` or any other variation. I don't think that supporting arbitrary variations is a worthwhile complexity vs ergonomics tradeoff.